src/shogun/machine/LinearMachine.cpp

```diff
@@ -120,3 +141,52 @@ void CLinearMachine::store_model_features()
 {
 }
 
+float64_t CLinearMachine::compute_bias(CFeatures* data)
+{
+    REQUIRE(m_labels,"No labels set\n");
+
+    if (!data)
+    	data=features;
+
+    REQUIRE(data,"No features provided and no featured previously set\n");
+
+    REQUIRE(m_labels->get_num_labels() == data->get_num_vectors(),
+    	"Number of training vectors (%d) does not match number of labels (%d)\n",
+    	m_labels->get_num_labels(),data->get_num_vectors());
+
+    SGVector<float64_t> outputs = apply_get_outputs(data);
+
+    int32_t num_vec=data->get_num_vectors();
+
+    Map<VectorXd> eigen_outputs(outputs,num_vec);
+    Map<VectorXd> eigen_labels(((CRegressionLabels*)m_labels)->get_labels(),num_vec);
+
+    return (eigen_labels - eigen_outputs).mean() ;
```

Comment:
I guess this method should be void and set the bias internally, rather than returning it